icicle/appUtils/ntt/kernel_ntt.cu

```diff
@@ -501,6 +562,100 @@ namespace ntt {
     return CHK_LAST();
   }
 
+  // Fast-twiddles: 2N twiddles for forward, 2N for inverse
+  template <typename S>
+  __global__ void generate_basic_twiddles_fast_twiddles_mode(S basic_root, S* basic_twiddles)
+  {
+    S w0 = basic_root * basic_root;
+    S w1 = (basic_root + w0 * basic_root) * S::inv_log_size(1);
```

Comment:
I don't know if the runtime of this function is at all important but I think this line can be changed to
```c++
S w1 = S::div2(basic_root + w0 * basic_root);
```
which is also more readable imo. The same applies to the next line and the `generate_basic_twiddles_generic` method.

Reply:
wasn't aware of that function, could be fine too

Reply:
are you sure about that? It computes incorrectly.

Reply:
Ohh sorry, `div2` only works for even inputs currently. I can make it work for odd inputs too ofc if it's important to make this function faster (though it seems that it only gets called a couple of times during twiddle generation). Otherwise let's leave it as it is.